scripts/getPVAliases.py

```diff
@@ -0,0 +1,508 @@
+# -*- coding: utf-8 -*-
+"""
+A script for gathering alias associations from an IOC
+user: aberges
+"""
+###############################################################################
+#%% Imports
+###############################################################################
+
+import sys
+import re
+import os.path
+import argparse
+import json
+import copy
+import glob as gb
+from colorama import Fore, Style
+from prettytable import PrettyTable
+
+###############################################################################
+#%% Functions
+###############################################################################
+
+def simple_prompt(prompt: str, default: str = 'N'):
+    """
+    Simple yes/no prompt which defaults to 'N' = False, but can be changed.
+
+    """
+    while True:
+        p = input(prompt).strip().lower()
+        if p in ['']:
+            p = default.lower()
+        if p[0] == 'y':
+            result = True
+            break
+        if p.lower()[0] == 'n':
+            result = False
+            break
+        print('Invalid Entry. Please choose again.')
+    return result
+
+
+def request_file_dest(prompt: str,
+                      default: str=os.getcwd(),):
+    """
+    Requests the user for a destination to save a file. 
+    Tests if the resultant path exists and makes the directory if necessary.
+    
+    Parameters
+    ----------
+    prompt : str
+        Prompt to show the user
+    default: str
+        File destination to default to.
+        Default is '{current_dir}/tempt.txt'
+    Returns
+    -------
+    str
+        Path to file destination.
+    """
+    while True:
+        p = input(prompt+f'(default = {default}): ')
+        # Check for default
+        if p in ['']:
+            result = default
+        else:
+            result = p
+        confirm = simple_prompt('Is '
+                                + Fore.LIGHTYELLOW_EX + f'{result} '
+                                + Style.RESET_ALL
+                                + 'correct? (y/N): ')
+        if confirm is True:
+            break
+    return result
+
+
+def flatten_list(input_list: list):
+    """
+    Flatten a list of lists and find its unique values.
+    Note: loses order due to set() call.
+    """
+    _result = [e for l in input_list for e in l]
+    return list(set(_result))
+
+
+def search_file(file:str,output:list=None,patt:str=None,prefix:str='',
+                color_wrap:Fore=None):
```

Comment:
I'm noticing that you've type hinted all the arguments but none of the return values, and none of the lists have their containing types, and a lot of these are type hinted incorrectly (type str set to None here, for example). Are you interested in filling in the missing type information or were you just going for a light pass to help write the script?

Reply:
I can clean this up. I think I originally left some of these as default = `None` for catching missing args or forcing break points in earlier testing and forgot to revert. I should go around and fill in all the `list[str]` and `list[dict]` etc.

I keep `output : list[str] = None` for reconstructing the `grep_ioc` output that I then parse again, so maybe it's superfluous and I should be more clever about this.